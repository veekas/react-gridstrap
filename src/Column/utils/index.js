export const breakpoints = {
  xs: 0,
  sm: 320,
  md: 768,
  lg: 1024,
  xl: 1280,
};

export const allowedValues = [
  true,
  'auto',
  '1',
  '2',
  '3',
  '4',
  '5',
  '6',
  '7',
  '8',
  '9',
  '10',
  '11',
  '12',
];
