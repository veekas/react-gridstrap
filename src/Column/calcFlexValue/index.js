import { allowedValues, breakpoints } from '../utils';

const calcFlexValue = (sizes) => {
  let mediaQueries = 'flex: 1 0 0;';

  if (typeof sizes === 'object') {
    const sizesArray = Object.keys(sizes);

    sizesArray.forEach((size) => {
      if (
        Object.prototype.hasOwnProperty.call(breakpoints, size)
        && allowedValues.includes(`${sizes[size]}`)
      ) {
        const minWidthValue = breakpoints[size];
        const growValue = sizes[size];

        mediaQueries += `
          @media (min-width: ${minWidthValue}px) {
            flex: ${growValue} 0 0;
          }
        `;
      }
    });
  }

  return mediaQueries;
};

export default calcFlexValue;
