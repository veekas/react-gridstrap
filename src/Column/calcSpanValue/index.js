import { allowedValues, breakpoints } from '../utils';

const calcSpanValue = (sizes) => {
  let mediaQueries = '';

  if (typeof sizes === 'object') {
    const sizesArray = Object.keys(sizes);

    sizesArray.forEach((size) => {
      if (
        Object.prototype.hasOwnProperty.call(breakpoints, size)
        && allowedValues.includes(`${sizes[size]}`)
      ) {
        const minWidthValue = breakpoints[size];
        const spanValue = sizes[size] === 'auto' ? '1' : sizes[size];

        mediaQueries += `
          @media (min-width: ${minWidthValue}px) {
            grid-column-end: span ${spanValue};
          }
        `;
      }
    });
  }

  return mediaQueries;
};

export default calcSpanValue;
