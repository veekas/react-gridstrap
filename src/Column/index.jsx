import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import calcFlexValue from './calcFlexValue';
import { allowedValues } from './utils';

const Column = ({ children, className, xs, sm, md, lg, xl, ...props }) => {
  const ColumnDiv = styled('div')`
    box-sizing: border-box;
    position: relative;
    ${calcFlexValue({ xs, sm, md, lg, xl })}
  `;

  return (
    <ColumnDiv className={className} {...props}>
      {children}
    </ColumnDiv>
  );
}

const columnSizePropType = PropTypes.oneOf(allowedValues);

Column.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ])),
  xs: columnSizePropType,
  sm: columnSizePropType,
  md: columnSizePropType,
  lg: columnSizePropType,
  xl: columnSizePropType,
};

Column.defaultProps = {
  children: null,
  className: null,
  style: null,
  xs: null,
  sm: null,
  md: null,
  lg: null,
  xl: null,
};

export default Column;
