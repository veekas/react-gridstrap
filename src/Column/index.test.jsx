import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Column from '.';

describe('Column', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Column />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // it('matches snapshot when passed zero props', () => {
  //   const tree = renderer
  //     .create(<Column />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
