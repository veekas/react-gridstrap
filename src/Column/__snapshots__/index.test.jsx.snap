// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column matches snapshot when passed zero props 1`] = `
<div
  className="css-23py4l"
  style={null}
/>
`;
